packages/web/modals/profile.tsx

```diff
@@ -103,6 +137,133 @@ export const ProfileModal: FunctionComponent<
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, []);
 
+  useEffect(() => {
+    if (address.length > 0) {
+      setStargazeAddress(
+        Bech32Address.fromBech32(address, "osmo").toBech32("stars")
+      );
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [address]);
+
+  const fetchStargazeNFTs = async () => {
```

Comment:
Wrap these as callbacks so we don't need to omit deps from useEffects.

Could also pull all these queries into their own hook to clean up the profile a bit

Reply:
Updated the first part.

Reluctant to add a hook as it may not find use anywhere else in the code.

Reply:
I agree with your take @MightOfOaks. We will likely not reuse any of this anytime soon, so it should be as isolated as possible in the codebase